source/lib/course-search/update-course-storage.js

```diff
@@ -17,54 +17,54 @@ type TermInfoType = {
 	type: string,
 }
 
-export async function updateStoredCourses(): boolean {
+export async function updateStoredCourses(): Promise<boolean> {
 	RNFS.readDir(COURSE_STORAGE_DIR + '/terms').catch(() => {
 		RNFS.mkdir(COURSE_STORAGE_DIR + '/terms')
 	})
 	const outdatedTerms: Array<TermType> = await determineOutdatedTerms()
-	outdatedTerms.forEach(term => {
-		storeTermCoursesFromServer(term.path)
-	})
+	const promises = outdatedTerms.map(term =>
+		storeTermCoursesFromServer(term.path),
+	)
+	await Promise.all(promises)
```

Comment:
Might as well join these two together – 

```js
await Promise.all(outdatedTerms.map(term =>
	storeTermCoursesFromServer(term.path),
))
```

(since we don't need the resolved values anywhere)